scripts/gaussian_codegen/shared/gates.py

```
@@ -0,0 +1,144 @@
+    max_z_raw: int,
+    n: int = 100_000,
+    min_headroom_bits: int = MIN_HEADROOM_BITS,
+) -> tuple[int, int]:
+    """Prove the peak full-width `acc.mag * z.mag` Horner product over `[0, max_z_raw]`
+    clears `2^256` with at least `min_headroom_bits` to spare. The product grows
+    smoothly with z (peak near `max_z_raw`), so a coarse grid plus the top input
+    captures it. Returns `(peak_bits, headroom_bits)`; raises on insufficient
+    headroom."""
+    step = max(1, max_z_raw // n)
+    peak = 0
+    for z_raw in list(range(0, max_z_raw, step)) + [max_z_raw - 1]:
```

Ruff (0.15.20)

[warning] Consider `[*list(range(0, max_z_raw, step)), max_z_raw - 1]` instead of concatenation

Replace with `[*list(range(0, max_z_raw, step)), max_z_raw - 1]`

(RUF005)